ganttproject/src/biz/ganttproject/impex/csv/GanttCSVExport.java

```diff
@@ -102,25 +97,84 @@ public void save(OutputStream stream) throws IOException {
       format = format.withQuote(csvOptions.sSeparatedTextChar.charAt(0));
     }
 
-    CSVPrinter csvPrinter = new CSVPrinter(writer, format);
+    return format;
+  }
+
+  private SpreadsheetWriter getWriter(OutputStream stream, String formatName) throws IOException {
+    if (formatName != null && formatName.length() > 0) {
```

Comment:
> Do I refactore ExportToimage with enums?
Good ideain general but out of this pull request scope :)

>  if the enum is expanded...the case default will work

Remove the default and let either IDE check if switch is exhaustive. Or wait until Java compiler will become smart enough to check it :)

> and secondly, it can be null.

Right. Please add `Preconditions.checkNotNull(format)` :)